packages/react-components/src/components/xy-output-component.tsx

```diff
@@ -321,6 +326,75 @@ export class XYOutputComponent extends AbstractTreeOutputComponent<AbstractOutpu
         );
     }
 
+    drawD3Chart(): JSX.Element {
+        const chartHeight = parseInt(this.props.style.height.toString());
+        const chartWidth = this.getChartWidth();
+
+        if (this.state.xyData.labels?.length > 0) {
+            const data: any[] = [];
+
+            this.state.xyData?.datasets?.forEach((dSet: any) => {
+                const row: any = [];
+                if (this.isScatterPlot) {
+                    dSet.data.forEach((tupple: any) => {
+                        row.push({ xValue: tupple.x, yValue: tupple.y });
+                    });
+                } else {
+                    dSet.data.forEach((y: number, j: number) => {
+                        row.push({ xValue: this.state.xyData.labels[j], yValue: y });
+                    });
+                }
+                data.push(row);
+            });
+
+            const yScale = scaleLinear()
+                .domain([this.state.allMin, Math.max(this.state.allMax, 1)])
+                .range([chartHeight - this.margin.bottom, this.margin.top]);
+
+            const xDomain = this.state.xyData.labels.length - 1;
+
+            const xScale = scaleLinear()
+                .domain([this.state.xyData.labels[0], this.state.xyData.labels[xDomain]].map(Number))
+                .range([0, chartWidth]);
+
+            if (this.chartRef.current) {
+                const ctx = this.chartRef.current.getContext('2d');
+
+                // Fix blurred lines in retina displays
+                const dpr = window.devicePixelRatio;
+                this.chartRef.current.width = dpr * chartWidth;
+                this.chartRef.current.height = dpr * chartHeight;
+                this.chartRef.current.style.width = chartWidth + 'px';
+                this.chartRef.current.style.height = chartHeight + 'px';
+                ctx.scale(dpr, dpr);
+
+                // Bar chart
+                if (ctx) {
+                    ctx.clearRect(0, 0, chartWidth, chartHeight);
+                    ctx.save();
+                    data.forEach((row, i) => {
+                        ctx.fillStyle = this.state.xyData.datasets[i].borderColor;
+                        row.forEach((tupple: any) => {
+                            ctx.beginPath();
+                            ctx.fillRect(xScale(Number(tupple.xValue)), chartHeight, 2, - chartHeight + yScale(+tupple.yValue));
```

Comment:
This is where the problem happens, Number(tupple.xValues) creates loss of precision from the bigint value.

Not sure, maybe this can be fixed by using getXForTime() ?

Also, is it intentional or a typo the + in: yScale(+tupple.yValue) ?

Reply:
Using `getXForTime()` now.
The `+` is not a typo, it's a unary plus operator.